client/state/jetpack/modules/reducer.js

```diff
@@ -16,53 +15,53 @@ import {
 import { combineReducers } from 'calypso/state/utils';
 
 const createItemsReducer = ( active ) => {
-	return ( state, { siteId, moduleSlug } ) => {
-		return merge( {}, state, {
-			[ siteId ]: {
-				[ moduleSlug ]: {
-					active,
-				},
+	return ( state, { siteId, moduleSlug } ) => ( {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ moduleSlug ]: {
+				...state[ siteId ]?.[ moduleSlug ],
+				active,
 			},
-		} );
-	};
+		},
+	} );
 };
 
 const createItemsListReducer = () => {
-	return ( state, { siteId, modules } ) => {
-		return merge( {}, state, {
-			[ siteId ]: modules,
-		} );
-	};
+	return ( state, { siteId, modules } ) => ( {
+		...state,
+		[ siteId ]: modules,
+	} );
 };
 
 const createRequestsReducer = ( data ) => {
-	return ( state, { siteId, moduleSlug } ) => {
-		return merge( {}, state, {
-			[ siteId ]: {
-				[ moduleSlug ]: data,
-			},
-		} );
-	};
+	return ( state, { siteId, moduleSlug } ) => ( {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ moduleSlug ]: data,
+		},
+	} );
```

Comment:
Before this PR, `createRequestsReducer` added (merged) a property (like `activating`) to the existing `state[ siteId ][ moduleSlug ]` object. Now it replaces it, discarding the old `data`. Is that OK?

`createItemsListReducer` has a similar issue. After this patch, `JETPACK_MODULES_RECEIVE` action replaces the existing list rather than merging into it? Looking at tests, `modules` is an object, not an array.